ext/dcos-installer/dcos_installer/action_lib/__init__.py

```diff
@@ -164,22 +177,21 @@ def install_dcos(config, block=False, state_json_dir=None, hosts=[], async_deleg
         },
         'agent': {
             'tags': {'role': 'agent', 'dcos_install_param': 'slave'},
-            'hosts': hosts if hosts else config['agent_list']
+            'hosts': hosts if hosts else config.get('agent_list', [])
+        },
+        'public_agent': {
+            'tags': {'role': 'public_agent', 'dcos_install_param': 'slave_public'},
+            'hosts': hosts if hosts else config.get('public_agent_list', [])
```

Comment:
This seems to say that if "hosts" is supplied we should always make them public_agents as well as private_agents